### PR TITLE
Revert vector type aliases to NumPy ndarrays

### LINKED ISSUE #3563: Multiplication of e.g. DOWN by a scalar is a type error on unreleased main branch
## Description of bug / unexpected behavior
Multiplication of `DOWN` by a float is a type error on `main`

image

This is because `DOWN` is a `Vector3D` and on `main` this is a union type including a tuple type:

```python
Vector3D: TypeAlias = Point3D
Point3D: TypeAlias = Union[InternalPoint3D, tuple[float, float, float]]
```

whereas on `v0.18.0` it is a `Vector3`:

```python
Vector3: TypeAlias = npt.NDArray[PointDType]
```

## Expected behavior
Multiplication of `DOWN` by a float should not be a type error.

## System specifications
VSCode Insiders Version: 1.86.0-insider
Commit: 9621add46007f7a1ab37d1fce9bcdcecca62aeb0

### PR DESCRIPTION
**Pull request by chopan050**
In the typing update #3484, the vector type aliases were redefined as unions between NumPy ndarrays and tuples, in an attempt to make them more consistent with the point type aliases.

However, this caused the issue #3563.

After some discussion, it was clarified that that change was unnecessary and incorrect, so this PR reverts the vector type aliases to only NumPy ndarrays.

Closes #3563

### CODE REVIEW

File: manim/typing.py [hunk 1]
```diff
@@ -332,7 +332,7 @@
     VMobjects!
 """
 
-Vector2D_Array: TypeAlias = Point2D_Array
+Vector2D_Array: TypeAlias = npt.NDArray[PointDType]
```
Comment (JasonGrace2282) [inline manim/typing.py:335]:
```suggestion
Vector2D_Array: TypeAlias = npt.NDArray[Vector2D]
```

File: manim/typing.py [hunk 2]
```diff
@@ -351,13 +351,13 @@
     VMobjects!
 """
 
-Vector3D_Array: TypeAlias = Point3D_Array
+Vector3D_Array: TypeAlias = npt.NDArray[PointDType]
```
Comment (JasonGrace2282) [inline manim/typing.py:354]:
```suggestion
Vector3D_Array: TypeAlias = npt.NDArray[Vector3D]
```

File: manim/typing.py [hunk 3]
```diff
@@ -368,19 +368,19 @@
     collisions.
 """
 
-VectorND_Array: TypeAlias = Union[npt.NDArray[PointDType], tuple[VectorND, ...]]
+VectorND_Array: TypeAlias = npt.NDArray[PointDType]
```
Comment (JasonGrace2282) [inline manim/typing.py:371]:
```suggestion
VectorND_Array: TypeAlias = npt.NDArray[VectorND]
```

**Review by JasonGrace2282 — CHANGES_REQUESTED** [review]
Thanks for making this PR!

File: manim/typing.py (hunk 1 above)
Comment (chopan050) [reply manim/typing.py:335]: I'm not sure about this change - when I wrote "An array of `Vector2D` objects", I meant something like "if you index a `Vector2D_Array` with a single index, what you get is a `Vector2D`", not _literally_ "this is an `(M,)`-shaped ndarray containing M `Vector2D` objects", which `npt.NDArray[Vector2D]` would seem to imply. Using `npt.NDArray[PointDType]` would refer to any NumPy ndarray whose elements are floats, in this case ndarrays of shape `(M, 2)`.

I'm not sure whether `npt.NDArray[Vector2D]` would be valid or correct in the first place - there can be ndarrays whose dtype is simply `object`, and one could have an `(M,)`-shaped ndarray with M objects - but this is not what is intended here :thinking:

Comment (chopan050) [reply manim/typing.py:335]: Although I would agree with this change in another context, the NumPy documentation seems to suggest using scalar types for dtypes, so `Vector2D` shouldn't be used as the dtype:

https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray

**Review by JasonGrace2282 — APPROVED** [review]
Looks good to me!

As a note for others, we discussed on discord that each element of a `VectorND_Array` is in fact a `VectorND`: however, the `dtype` of said `VectorND_Array` is given by numpy to be `float64` so that's why `VectorND_Array` (and similar aliases) are typed as `NDArray[PointDType]`

**Pull request merged**